hera_sim/visibilities/fftvis.py

```diff
@@ -316,3 +310,42 @@ def simulate(self, data_model):
             return 0
         else:
             return visfull
+
+    def _reorder_vis(self, req_pols, uvdata, visfull, vis, antpairs, polarized):
+        try:
+            if getattr(uvdata, "blt_order", None)[0] == "time":
```

Comment:
this can now be written as `if not uvdata.time_axis_faster_than_bls:`

Reply:
I think, actually, that this function can be a lot simpler... since `FFTVis` _only_ works for rectangular blts (you check this in the validation method), you should be able to do something like the following, for the whole function. This is because you're certain to have data coming out of FFTVis that has the same antpair ordering as the UVData (since you pass in the antpairs from the UVData to the simulation function)

```python
if uvdata.time_axis_faster_than_bls:
    vis = vis.transpose((3, 0, 1, 2))
else:
    vis = vis.transpose((0, 3, 1, 2))
 
if polarized:
    for p, (p1, p2) in enumerate(req_pols):
        visfull[:, p] = vis[:, p1, p2].reshape(-1)
    else:
        visfull[:, 0] = vis.reshape(-1)
```

Note that I think it would be even _better_, if possible, for the pols from the uvdata to be passed to the `simulate()` function, so the pols are automatically arranged in the same order as the uvdata, and also if possible for the antpairs axis in the `fftvis` output to come _before_ the pols (so you pass out something in the shape `(ntimes, nantpairs, npols)` OR `(ntimes, nantpairs, nfeed, nfeed)`).  That would require no reordering here. 

We can't do this in matvis because the structure of the outer product naturally puts the output into the form `(ntimes, nfeed, nfeed, nant, nant)`. But I think you could do this in `fftvis`.

Reply:
Thanks for the suggestion! This works great for most of the tests, but fails when `uvdata.reorder_blts(order='ant2')` is used. It seems like the reason for the failure came from my assumption that the order of `uvdata.get_antpairs()` reflected the baseline order in the data array, when that turned out not to be the case. I have a fix which gets the correct order of the baselines in the data and uses your solution. It may be a little hacky, so if there's a better way to can think to do this, I'd be happy to change it!

Reply:
Got it -- I should have noticed that actually. Yeah `get_antpairs()` is not guaranteed to return antpairs in any given order. I made a suggestion (see below) which should be both faster and less hacky -- and in our case, always correct. Feel free to assign `data_model.uvdata` to some shorter variable name (e.g. `uvd`) higher up in the script to enhance readability...